File: TypeScript/Easy/108.ts
// Daily Coding Problem: 108
// Difficulty: Easy
// Asked by: Google
// Authored by: TenthEdict

function checkStringRotation(a: string, b: string): boolean {
    if (a.length != b.length) { return false }
    return (a+a).includes(b);
}

console.log(checkStringRotation("abcde", "cdeab"));
console.log(checkStringRotation("abcde", "abcde"));
console.log(checkStringRotation("abc", "acb"));
console.log(checkStringRotation("abcde", "abcd"));
console.log(checkStringRotation("", ""));
console.log(checkStringRotation("a", "a"));

File: Go/Easy/108.go
// Daily Coding Problem: 108
// Difficulty: Easy
// Asked by: Google
// Authored by: TenthEdict

package main

import (
	"fmt"
	"strings"
)

func checkStringRotation(a string, b string) bool {
	if len(a) != len(b) { return false}
	return strings.Contains(a+a, b)
}

func main() {
	fmt.Println(checkStringRotation("abcde", "cdeab"))
	fmt.Println(checkStringRotation("abc", "acb"))    // false
	fmt.Println(checkStringRotation("", ""))          // true
	fmt.Println(checkStringRotation("abc", "abcd"))   // false
}